common/src/main/java/com/microsoft/identity/common/internal/dto/AccessTokenRecord.java

```diff
@@ -295,6 +310,24 @@ public void setExpiresOn(final String expiresOn) {
         mExpiresOn = expiresOn;
     }
 
+    /**
+     * Gets the refresh_on timestamp.
+     *
+     * @return The refresh_on to get.
+     */
+    public String getRefreshOn() {
+        return mRefreshOn;
+    }
+
+    /**
+     * Sets the refresh_on timestamp.
+     *
+     * @param refreshOn The refresh_on to set.
+     */
+    public void setRefreshOn(final String refreshOn) {
+        mRefreshOn = refreshOn;
+    }
+
     private boolean isExpired(final String expires) {
```

Comment:
Please let me know if you'd like me to add a shouldRefresh function, based on isExpired.

C++ doesn't use the isExpired logic (we've got our own logic in a higher, x-plat layer), but if Android plans to support refresh_in anytime soon, I'm happy to lay the groundwork as part of this PR :)

Reply:
@logdog16 If you can add tests for it, I wouldn't decline having this API. I'm assuming it's 1 additional method?